tests/test_validation_pandapipes.py

```diff
@@ -0,0 +1,268 @@
+
+
+class ValidateWithPandaPipes(TestCase):
+    """
+    Test case for a heat network consisting out of a source, pipe(s) and a sink
+    """
+
+    def test_heat_network_head_loss(self):
+        """
+        Heat network: compare the piecewise linear inequality constraints of the head loss
+        approximation to pandapipes
+
+        Checks:
+        - head losses
+        - cp value
+        """
+        root_folder = str(Path(__file__).resolve().parent.parent)
+        sys.path.insert(1, root_folder)
+
+        import examples.pandapipes.src.run_example
+        from examples.pandapipes.src.run_example import HeatProblemHydraulic
+
+        base_folder = Path(examples.pandapipes.src.run_example.__file__).resolve().parent.parent
+
+        class SourcePipeSinkNetwork(HeatProblemHydraulic):
+
+            def energy_system_options(self):
+                options = super().energy_system_options()
+                self.heat_network_settings["head_loss_option"] = (
+                    HeadLossOption.LINEARIZED_N_LINES_WEAK_INEQUALITY
+                )
+                self.heat_network_settings["n_linearization_lines"] = 10
+
+                return options
+
+            def times(self, variable=None) -> np.ndarray:
+                """
+                Shorten the timeseries to speed up the test
+
+                Parameters
+                ----------
+                variable : string with name of the variable
+
+                Returns
+                -------
+                The timeseries
+                """
+                return super().times(variable)[:9]
+
+        demand_time_series_file = "timeseries_constant.csv"
+
+        solution = run_esdl_mesido_optimization(
+            SourcePipeSinkNetwork,
+            base_folder=base_folder,
+            esdl_file_name="Test_Tree_S1C1.esdl",
+            esdl_parser=ESDLFileParser,
+            profile_reader=ProfileReaderFromFile,
+            input_timeseries_file=demand_time_series_file,
+        )
+        results = solution.extract_results()
+
+        demand_matching_test(solution, results)
+        energy_conservation_test(solution, results)
+        heat_to_discharge_test(solution, results)
+
+        # ------------------------------------------------------------------------------
+        # ------------------------------------------------------------------------------
+        # Setup and run pandapipes
+        # ------------------------------------------------------------------------------
+        # ------------------------------------------------------------------------------
+        from examples.pandapipes.src.pandapipeesdlparser import PandapipeEsdlParserClass
+
+        esdl_file = os.path.join(base_folder, "model", "Test_Tree_S1C1.esdl")
+        esdlparser = PandapipeEsdlParserClass()
+        esdlparser.loadESDLFile(esdl_file)
+        esdlparser.add_additional_joint()
```

Comment:
maybe a comment on why this line of code is needed

Reply:
Agree. Comment has been added